src/main/webapp/availability.js

```diff
@@ -12,86 +12,103 @@
-            timeslotsContainer.appendChild(errorMessage);
+        // Don't create a calendar if there are no available timeslots
+        if (timeslots === undefined || timeslots.length == 0) {
             return;
         }
-    });
-}
 
-function createTimeSlotBox(timeslot, tutorID) {
-    var months = [ "January", "February", "March", "April", "May", "June", 
-           "July", "August", "September", "October", "November", "December" ];
+        // There are available timeslots, display header and calendar
+        document.getElementById('calendar-header').style.display = 'block';
+        
+        const container = document.getElementById('calendar');
+        const chart = new google.visualization.Timeline(container);
 
-    const timeslotElement = document.createElement('li');
-    timeslotElement.className = 'list-group-item';
+        const dataTable = new google.visualization.DataTable();
+        dataTable.addColumn({type: 'string', id: 'Date'});
+        dataTable.addColumn({type: 'date', id: 'Start'});
+        dataTable.addColumn({type: 'date', id: 'End'});
 
-    const dateElement = document.createElement('h3');
-    dateElement.style.textAlign = 'left';
-    dateElement.style.display = 'inline';
-    var hour = Math.floor(parseInt(timeslot.start) / 60);
-    var amOrPm = "am";
-    if (hour > 12) {
-        hour = hour - 12;
-        amOrPm = "pm"
-    }
-    var minute = parseInt(timeslot.start) % 60;
-    if (minute == 0) {
-        minute = "00";
-    }
-    dateElement.innerText = hour + ":" + minute + amOrPm + " on " + months[timeslot.date.month] + " " + timeslot.date.dayOfMonth + ", " + timeslot.date.year;
+        // Sort timeslots in chronological order
+        timeslots.sort(function(a, b) {
+            return (a.date.year + a.date.month/12 + a.date.dayOfMonth/365) - (b.date.year + b.date.month/12 + b.date.dayOfMonth/365);
+        });
 
-    const dateLineElement = document.createElement('div');
-    dateLineElement.className = 'd-flex w-100 justify-content-between';
-    dateLineElement.style.padding = '10px';
-    dateLineElement.appendChild(dateElement);
+        for (var slot of timeslots) {
+            // Add 1 to the month so it displays correctly (January's default value is 0, February's is 1, etc.)
+            var date = (slot.date.month+1) + '/' + slot.date.dayOfMonth + '/' + slot.date.year;
+            dataTable.addRow([
+                date, asDate(slot.start), asDate(slot.end)
+            ]);
+        }
 
-    const selectButtonElement = document.createElement('button');
-    selectButtonElement.innerText = 'Select';
-    selectButtonElement.style.textAlign = 'right';
-    selectButtonElement.style.display = 'inline';
-    selectButtonElement.className = 'btn btn-default btn-lg';
-    selectButtonElement.addEventListener('click', () => {
-        selectTimeSlot(tutorID, window, timeslot);
-    });
+        // Have timeline span 24 hours regardless of what's scheduled
```

Comment:
Does this show the entire day now instead of just the scheduled session?

Reply:
Yup